Add CentOS / RHEL 8 support

alexandre-allard opened the pull request:
**Component**: salt, build, packages

**Context**: 
Support for CentOS 8 / RHEL 8

**Summary**:

**Acceptance criteria**:
Installation on RHEL 8 works (single-node-install-rhel-8) and on other dists also.

---

Closes: #2164

alexandre-allard committed:
scripts: Fail hard if missing packages during bootstrap

Add the skip_missing_names_on_install to yum install
to exit in error if one of the packages is missing,
but it should never happen since we embed these
packages.

alexandre-allard committed:
vagrant: Add support for CentOS 8

We can now spawn VM on CentOS 8

alexandre-allard committed:
build, packages: handle different RedHat release

Create a tree under packages/redhat to handle
multiple releases.
Adapt paths in the buildchain to the new scheme.

Refs: #2164

alexandre-allard committed:
build, packages: handle different Ubuntu releases

Create a tree under `packages/debian` to handle
multiple releases.
Adapt paths in the buildchain to the new scheme.

Refs: #2164

alexandre-allard committed:
salt,scripts: remove python36-rpm install

This is no longer needed as it has been fixed
in Salt 3002, python36-rpm is now a direct
dependency of salt-minion.

alexandre-allard committed:
scripts: Add dist release in repositories path

Since we now support multiple release, we need
to append the release in the path to retrieve
the right repository.

Refs: #2164

alexandre-allard committed:
packages: Remove warning when using apt in script

Apt is not made to be used in a non-interactive way,
but we do in the script used to fetch and build
ubuntu packages.
We deactivate the warning as it pollutes log output
when there is a need to debug.

alexandre-allard committed:
packages: Small fix of awk script

Use match to retrieve matched values instead of
doing a match and then a gensub.

alexandre-allard committed:
packages: Filter out SRPMs from repoquery output

It looks like on CentOS 8, repoquery returns some
SRPMs when looking for a package dependency.
So, let's filter them out since we don't care about
these.

Refs: #2164

alexandre-allard committed:
packages: Add RHEL 8 package builder Dockerfile

This is the definition to build the container
image used to download and build MetalK8s packages
and dependencies for RHEL 8.

Refs: #2164

alexandre-allard committed:
salt: Fix repoquery option order for RHEL 8

In RHEL 8, the package name in repoquery command
is no longer a positional argument, but the
argument of the `--whatrequires` option, so we
need to move this option right after the package
name.

Refs: #2164

alexandre-allard committed:
packages: Build metalk8s-sosreport for Python3

Since RHEL 8 does not ship Python 2, we build
this package for Python 3 instead 2.

Refs: #2164

alexandre-allard committed:
packages: Disable containerd btrfs support

btrfs is no longer supported on RHEL 8, so we
disable btrfs support for containerd on RHEL 8.

Refs: #2164

alexandre-allard committed:
build, packages: Add support for CentOS 8

Download and build packages for CentOS 8

Refs: #2164

alexandre-allard committed:
build: Add builder for RHEL 8

The purpose of this builder is to download
and build RHEL 8 packages.

Refs: #2164

alexandre-allard committed:
scripts, salt: Add support for RHEL 8

Refs: #2164

alexandre-allard committed:
vagrant: Add support for RHEL 8

Refs: #2164

alexandre-allard committed:
salt: Add --quiet option to repoquery

We need this option on RHEL 8, otherwise we end
up with RHSM information polluting stdout which
then breaks the module waiting for a specific
output format `<package-name> <version>`.

Refs: #2164

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

alexandre-allard committed:
eve: Replace RHEL 7 single node with RHEL 8

Since we now support RHEL 8, we want to test it
in pre-merge on a single-node instead of RHEL 7.

Refs: #2164

NicolasT commented on buildchain/buildchain/packaging.py:
```diff
@@ -181,269 +288,381 @@ def clean() -> None:
 
 # Calico Container Network Interface Plugin.
-CALICO_RPM = _rpm_package(
+REDHAT_7_CALICO_RPM = _rpm_package(
```
There's a difference between how we handle the `7` and `8` package for Calico. Would it make sense to turn this into a function taking a `releasever`, then call it twice in `RPM_TO_BUILD` construction?

alexandre-allard replied:
Totally

NicolasT commented on buildchain/buildchain/packaging.py:
```diff
@@ -181,269 +288,381 @@ def clean() -> None:
+                ],
+            ),
+            REDHAT_7_CALICO_RPM,
+            _rpm_package(
```
Some duplication between 7 and 8 here as well. Could this be lifted?

NicolasT commented on buildchain/buildchain/packaging.py:
```diff
@@ -181,269 +288,381 @@ def clean() -> None:
-    ),
+        '8': (
+            # SOS report custom plugins.
+            _rpm_package(
```
Same.

NicolasT commented on buildchain/buildchain/versions.py:
```diff
@@ -21,7 +21,6 @@
 K8S_VERSION        : str = '1.18.13'
 SALT_VERSION       : str = '3002.2'
 CONTAINERD_VERSION : str = '1.4.3'
-CONTAINERD_RELEASE : str = '1.el7'
```
Why would this be gone?

NicolasT commented on buildchain/buildchain/versions.py:
```diff
@@ -336,78 +336,121 @@ def deb_full_name(self) -> str:
+            PackageVersion(
+                name='calico-cni-plugin',
+                version=CALICO_VERSION,
+                release='1.el7'
```
We shouldn't have this hard-coded but somewhere close to `CALICO_VERSION`.

NicolasT commented on buildchain/buildchain/versions.py:
```diff
@@ -336,78 +336,121 @@ def deb_full_name(self) -> str:
+            PackageVersion(
+                name='containerd',
+                version=CONTAINERD_VERSION,
+                release='2.el7',
```
Not coded up here. This is a variable.

NicolasT commented on buildchain/buildchain/versions.py:
```diff
@@ -336,78 +336,121 @@ def deb_full_name(self) -> str:
+            PackageVersion(
+                name='metalk8s-sosreport',
+                version=SHORT_VERSION,
+                release='2.el7'
```
Same.

NicolasT commented on buildchain/buildchain/versions.py:
```diff
@@ -336,78 +336,121 @@ def deb_full_name(self) -> str:
+            PackageVersion(
+                name='calico-cni-plugin',
+                version=CALICO_VERSION,
+                release='1.el8'
```
Same

NicolasT commented on buildchain/buildchain/versions.py:
```diff
@@ -336,78 +336,121 @@ def deb_full_name(self) -> str:
+            PackageVersion(
+                name='containerd',
+                version=CONTAINERD_VERSION,
+                release='2.el8',
```
Same

NicolasT commented on buildchain/buildchain/versions.py:
```diff
@@ -336,78 +336,121 @@ def deb_full_name(self) -> str:
+            PackageVersion(
+                name='metalk8s-sosreport',
+                version=SHORT_VERSION,
+                release='2.el8'
```
Same

NicolasT commented on buildchain/buildchain/versions.py:
```diff
@@ -336,78 +336,121 @@ def deb_full_name(self) -> str:
+            PackageVersion(
+                name='calico-cni-plugin',
+                version=CALICO_VERSION,
+                release='1'
```
Same, though this is *not* the same `release` as the RPM `release`...

NicolasT commented on buildchain/buildchain/versions.py:
```diff
@@ -336,78 +336,121 @@ def deb_full_name(self) -> str:
+            PackageVersion(
+                name='metalk8s-sosreport',
+                version=SHORT_VERSION,
+                release='1'
```
Same as previous Ubuntu comment.

NicolasT commented on packages/redhat/7/Dockerfile:
```diff
@@ -19,9 +20,8 @@ RUN yum install -y \
-RUN yum install -y \
+RUN yum install -y --setopt=skip_missing_names_on_install=False \
         golang \
-        btrfs-progs-devel \
```
This is quite surprising?

NicolasT commented on packages/redhat/8/Dockerfile:
```diff
@@ -0,0 +1,34 @@
+# Equal to centos:8.2.2004
```
This file could/should be more aligned with the RHEL7 one

NicolasT commented on packages/redhat/8/Dockerfile:
```diff
@@ -0,0 +1,34 @@
+
+ENV GOPATH=/go
+
+RUN dnf install -y --setopt=skip_missing_names_on_install=False \
```
Why not have these with the previous `dnf install`?

NicolasT commented on packages/redhat/common/containerd.spec:
```diff
@@ -23,16 +23,16 @@ export GOPATH=$GO_BUILD_PATH:%{gopath}
 %global _dwz_low_mem_die_limit 0
 %ifnarch ppc64
-go build -buildmode pie -compiler gc -tags="rpm_crashtraceback ${BUILDTAGS:-seccomp}" -ldflags "${LDFLAGS:-} -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '%__global_ldflags %{?__golang_extldflags}'" -a -v -x %{?**};
+go build -buildmode pie -compiler gc -tags="rpm_crashtraceback ${BUILDTAGS:-seccomp no_btrfs}" -ldflags "${LDFLAGS:-} -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '%__global_ldflags %{?__golang_extldflags}'" -a -v -x %{?**};
```
The `no_btrfs` should be releasever dependent. Enabled on EL7 as was the case before, disabled on EL8.

NicolasT commented on packages/redhat/common/containerd.spec:
```diff
@@ -23,16 +23,16 @@ export GOPATH=$GO_BUILD_PATH:%{gopath}
+go build -buildmode pie -compiler gc -tags="rpm_crashtraceback ${BUILDTAGS:-seccomp no_btrfs}" -ldflags "${LDFLAGS:-} -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '%__global_ldflags %{?__golang_extldflags}'" -a -v -x %{?**};
 %else
-go build -compiler gc -tags="rpm_crashtraceback ${BUILDTAGS:-seccomp}" -ldflags "${LDFLAGS:-} -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '%__global_ldflags %{?__golang_extldflags}'" -a -v -x %{?**};
+go build -compiler gc -tags="rpm_crashtraceback ${BUILDTAGS:-seccomp nobtrfs}" -ldflags "${LDFLAGS:-} -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '%__global_ldflags %{?__golang_extldflags}'" -a -v -x %{?**};
```
Same

NicolasT commented on packages/redhat/common/containerd.spec:
```diff
@@ -43,8 +43,9 @@ Source2:        containerd.toml
 
 BuildRequires:  golang >= 1.10
-BuildRequires:  btrfs-progs-devel
+%if 0%{?el7}
```
Now the EL8 build is not self-hosted, which I think we should fix.

NicolasT commented on packages/redhat/common/containerd.spec:
```diff
@@ -219,6 +220,9 @@ install -D -p -m 0644 %{S:2} %{buildroot}%{_sysconfdir}/containerd/config.toml
 
 %changelog
+* Tue Dec 8 2020 Alexandre Allard <alexandre.allard@scality.com> - 1.4.3-2
+- Disable btrfs support
```
Right. Not super fond though.

Also, `Btrfs`.

NicolasT commented on packages/redhat/common/metalk8s-sosreport.spec:
```diff
@@ -0,0 +1,43 @@
+ExclusiveArch:  x86_64
+ExclusiveOS:    Linux
+
+License:        GPL+
```
Is it?

alexandre-allard replied:
I'm not really familiar with licensing stuff and it's not related to that PR, but as far as I understand, it is.

sosreport report is distributed under GNU GPL license which is a copyleft license:
https://github.com/sosreport/sos/blob/master/LICENSE

```
The GPL series are all copyleft licenses, which means that any derivative work must be distributed under the same or equivalent license terms.
```
source: https://en.wikipedia.org/wiki/GNU_General_Public_License

If it's not, could you explain why ?

NicolasT replied:
`sosreport`/`sos` is GPL. Our plugin is likely not a 'derived work', and all code in MetalK8s is APL2-licensed. So I'd argue our SOS plugin should also be APL2 licensed?

NicolasT commented on salt/metalk8s/map.jinja:
```diff
@@ -2,38 +2,77 @@
+  salt['grains.filter_by']({
+    'RedHat': {
+      '7': {
+        'python3': 'python36-rpm',
```
This is a bit confusing to me.

alexandre-allard replied:
Yes, it is really confusing, because it's a mistake. :)

NicolasT commented on salt/metalk8s/map.jinja:
```diff
@@ -56,111 +95,163 @@
+    'RedHat': {
+      '7': {
+        'containerd': {
+          'name': 'epel'
```
`containerd` is no longer the one from `epel`.

alexandre-allard replied:
Yes, didn't change anything, it was already there, but those entries (`repo.containerd` and `repo.kubernetes`) are leftovers and should be cleaned up. I'll remove them.

NicolasT commented on salt/metalk8s/map.jinja:
```diff
@@ -56,111 +95,163 @@
-          'trusted': 'yes',
+      '8': {
+        'containerd': {
+          'name': 'epel'
```
Same

NicolasT commented on salt/metalk8s/repo/redhat.sls:
```diff
@@ -13,9 +13,15 @@ Install yum-utils:
+{%- if grains.osmajorrelease < 8 %}
+Install versionlock plugin:
   pkg.installed:
     - name: yum-plugin-versionlock
```
Why don't we do this through the package name per OS map thing we already have in place?

alexandre-allard replied:
We totally could, I'll change it.

NicolasT commented on scripts/bootstrap.sh.in:
```diff
@@ -55,11 +55,20 @@ BASE_DIR=$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")
+# shellcheck disable=SC2034
+# Used through an indirection
+declare -a PACKAGES_REDHAT_7=(
     python36-rpm
```
Whenever I see `python36-rpm` for EL7 and `python36` for EL8 I'm very confused. `python36-rpm` is not alike `python36`, right?

TeddyAndrieux replied:
No, python36-rpm is the python36 lib to interact with rpm
Maybe it's not needed in rhel8 (and btw it's maybe also no longer needed with salt 3002.2 we have now, I didn't checked, but it's not related to this PR btw https://github.com/saltstack/salt/issues/57972)
I agree it's really different stuff and to be consistent we may want to install "python36-rpm" on rhel8 as well (if this package exists, otherwise maybe we should add also "python36" in the list for rhel7 even if it's not really needed since it's a dependency of "python36-rpm")

NicolasT replied:
Exactly. So, if we need `python36-rpm` on RHEL7 systems *and* `python36` (when not using the RPM bindings), we should declare `python36` as a dependency, and not rely on it being implicitly available because we install `python36-rpm` which has `python36` as a dep.
If on RHEL8 we *don't* need `python36-rpm`, then it shouldn't be listed and only include `python36` (or similar).

Let's try to keep the differences between the OSes as minimal as possible.

TeddyAndrieux commented on scripts/common.sh:
```diff
@@ -103,19 +105,19 @@ EOF
 
 get_packages_list() {
-    python -c "
+    python3 -c "
```
It cannot works on rhel-7 since python3 may not be installed when we run this command

NicolasT replied:
Similarly, I seem to remember RHEL8 could come with no `python` pre-installed (except for the non-system-one installed for `dnf` to function) at all?

alexandre-allard committed:
packages: Change metalk8s-sosreport licensing

Since all code in MetalK8s is APL2 licensed,
let's put our sosreport plugin under the same
license

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/2164-support-centos-8` into
`development/2.7`
* [w/2.8/feature/2164-support-centos-8](https://github.com/scality/metalk8s/compare/development/2.8...w/2.8/feature/2164-support-centos-8) will be merged into `development/2.8` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

bert-e commented:
# History mismatch



Merge commit #1b51240e40a36dca6f7908a4ccd9686ef89c2895 on the integration branch
`w/2.8/feature/2164-support-centos-8` is merging a branch which is neither the current
branch `feature/2164-support-centos-8` nor the development branch
`development/2.8`.

It is likely due to a rebase of the branch `feature/2164-support-centos-8` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard committed:
salt: Remove useless entries in map

Clean up entries that are no longer used
anywhere. These are leftovers when we were
using fromrepo to install embedded packages.

bert-e commented:
# History mismatch



Merge commit #376abd07345365be5aa2dbbfe7b5a8ce3de3776d on the integration branch
`w/2.8/feature/2164-support-centos-8` is merging a branch which is neither the current
branch `feature/2164-support-centos-8` nor the development branch
`development/2.8`.

It is likely due to a rebase of the branch `feature/2164-support-centos-8` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard committed:
scripts: install python3 during bootstrap

On CentOS/RHEL 8, ensure Python3 is installed
prior to do anything else.
This is needed because there is no Python installed
otherwise and we need it for some scripts
(e.g. to check packages presence, before proceeding
to bootstrap)

Refs: #2164

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/2164-support-centos-8` into
`development/2.7`
* [w/2.8/feature/2164-support-centos-8](https://github.com/scality/metalk8s/compare/development/2.8...w/2.8/feature/2164-support-centos-8) will be merged into `development/2.8` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

bert-e commented:
# History mismatch



Merge commit #ebfb1157ae2a9ca5c87b54362ce1607a7e6bf3d8 on the integration branch
`w/2.8/feature/2164-support-centos-8` is merging a branch which is neither the current
branch `feature/2164-support-centos-8` nor the development branch
`development/2.8`.

It is likely due to a rebase of the branch `feature/2164-support-centos-8` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# History mismatch



Merge commit #0879315b1d49ca58f6aa6bc03f76bd0bd7860ca5 on the integration branch
`w/2.8/feature/2164-support-centos-8` is merging a branch which is neither the current
branch `feature/2164-support-centos-8` nor the development branch
`development/2.8`.

It is likely due to a rebase of the branch `feature/2164-support-centos-8` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

alexandre-allard committed:
Add CHANGELOG entry for CentOS/RHEL 8 support

Refs: #2164

bert-e commented:
# Conflict




There is a conflict between your branch `feature/2164-support-centos-8` and the
destination branch `development/2.8`.

Please resolve the conflict on **the feature branch** (`feature/2164-support-centos-8`).

```bash
 $ git fetch
 $ git checkout origin/feature/2164-support-centos-8
 $ git merge origin/development/2.8
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:feature/2164-support-centos-8
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

TeddyAndrieux commented on CHANGELOG.md:
```diff
@@ -2,6 +2,8 @@
 ## Release 2.8.0 (in development)
 ### Enhancements
+- Add CentOS/RHEL 8 support
+
```
Officially we only support RHEL 8 no ? so I think we shouldn't talk about CentOS in this changelog (and also maybe in the PR title)

You also need to add a link to the PR and the issue

TeddyAndrieux commented on eve/workers/openstack-terraform/terraform/nodes.tf:
```diff
@@ -157,23 +157,23 @@ resource "openstack_compute_instance_v2" "bootstrap" {
     inline = [
-      "if [ '${var.os}' = 'rhel-7' ]; then sudo chmod +x scripts/rhsm-register.sh; fi",
-      "if [ '${var.os}' = 'rhel-7' ]; then sudo scripts/rhsm-register.sh '${var.rhsm_username}' '${var.rhsm_password}'; fi"
+      "case '${var.os}' in rhel-*) sudo chmod +x scripts/rhsm-register.sh;; esac",
+      "case '${var.os}' in rhel-*) sudo scripts/rhsm-register.sh '${var.rhsm_username}' '${var.rhsm_password}';; esac"
```
Just a question, does it bring any value to use "case" here instead of the "old" if with a wildcare ?

alexandre-allard replied:
Nothing, but terraform is using good old sh, so no bashism is available hence the use of `case` to do some glob comparison.
I could have used a `if [ "${var##pattern}" != "$var" ]; then ...; fi` construction also, but I find this one easier to read and it would need an extra step to store `${var.os}` in a shell var first.

TeddyAndrieux reviewed (commented):
Minor comments otherwise LGTM

TeddyAndrieux reviewed (dismissed):
LGTM

bert-e commented:
# Conflict




There is a conflict between your branch `feature/2164-support-centos-8` and the
destination branch `development/2.8`.

Please resolve the conflict on **the feature branch** (`feature/2164-support-centos-8`).

```bash
 $ git fetch
 $ git checkout origin/feature/2164-support-centos-8
 $ git merge origin/development/2.8
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:feature/2164-support-centos-8
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

alexandre-allard commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality










*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.8`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.8`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
